comp/autoscaling/datadogclient/impl/fallbackclient.go

```diff
@@ -0,0 +1,153 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2021-present Datadog, Inc.
+
+package datadogclientimpl
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	"gopkg.in/zorkian/go-datadog-api.v2"
+
+	configComponent "github.com/DataDog/datadog-agent/comp/core/config"
+	httputils "github.com/DataDog/datadog-agent/pkg/util/http"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type datadogIndividualClient struct {
+	client             *datadog.Client
+	lastQuerySucceeded bool
+	lastFailure        time.Time
+	lastSuccess        time.Time
+	retryInterval      time.Duration
+}
+
+const (
+	minRetryInterval = 30 * time.Second
+	maxRetryInterval = 30 * time.Minute
+)
+
+// DatadogFallbackClient represents a datadog client able to query metrics to a second Datadog endpoint if the first one fails
+type datadogFallbackClient struct {
+	clients        []*datadogIndividualClient
+	lastUsedClient int
+}
+
+// NewDatadogFallbackClient generates a new client able to query metrics to a second Datadog endpoint if the first one fails
+func newDatadogFallbackClient(config configComponent.Component, endpoints []endpoint) (*datadogFallbackClient, error) {
+	if len(endpoints) == 0 {
+		return nil, log.Errorf("%s must be non-empty", metricsRedundantEndpointConfig)
```

Comment:
Let's take the logger component as a parameter and use it instead of `pkg/util/log`.